interceptor.go

```diff
@@ -0,0 +1,90 @@
+// +build !js
+
+package webrtc
+
+import (
+	"io"
+
+	"github.com/pion/rtcp"
+	"github.com/pion/rtp"
+)
+
+// WriteRTP is used by Interceptor.BindLocalTrack.
+type WriteRTP func(p *rtp.Packet, attributes map[interface{}]interface{}) (int, error)
+
+// ReadRTP is used by Interceptor.BindRemoteTrack.
+type ReadRTP func() (*rtp.Packet, map[interface{}]interface{}, error)
+
+// WriteRTCP is used by Interceptor.BindWriteRTCP.
+type WriteRTCP func(pkts []rtcp.Packet, attributes map[interface{}]interface{}) (int, error)
+
+// ReadRTCP is used by Interceptor.BindReadRTCP.
+type ReadRTCP func() ([]rtcp.Packet, map[interface{}]interface{}, error)
+
+// Interceptor can be used to add functionality to you PeerConnections by modifying any incoming/outgoing rtp/rtcp
+// packets, or sending your own packets as needed.
+type Interceptor interface {
+
+	// BindReadRTCP lets you modify any incoming RTCP packets. It is called once per sender/receiver, however this might
+	// change in the future. The returned method will be called once per packet batch.
+	BindReadRTCP(read ReadRTCP) ReadRTCP
+
+	// BindWriteRTCP lets you modify any outgoing RTCP packets. It is called once per PeerConnection. The returned method
+	// will be called once per packet batch.
+	BindWriteRTCP(write WriteRTCP) WriteRTCP
+
+	// BindLocalTrack lets you modify any outgoing RTP packets. It is called once for per LocalTrack. The returned method
+	// will be called once per rtp packet.
+	BindLocalTrack(ctx *TrackLocalContext, write WriteRTP) WriteRTP
```

Comment:
It looks very confusing with `context.Context`. `TrackLocalContext` may be better to be changed?

Reply:
I think we should define a new type just for interceptors. We might hit snags where we want to add things to Interceptors (and not Track)

Reply:
I think we won't be able to use it anyway once the dependency is reverted to webrtc -> pkg/interceptor